src/Marten/Events/Event.cs

```diff
@@ -72,7 +72,7 @@ public interface IEvent
 
     public interface IEvent<out T> : IEvent
     {
-        T GetData();
+        new T Data { get; }
```

Comment:
Does this actually work? If the test class `using_non_concrete_types_in_projections` is actually passing, I'm perfectly good with this PR.